### PR TITLE
Improve documentation and type specs

### PR DESCRIPTION
**Pull request by arkgil**
Improved docs and specs for client modules.

### CODE REVIEW

**Comment by arkgil** [comment]
@mentels  take a look at that please.

File: lib/jerboa/client.ex [hunk 1]
```diff
@@ -1,28 +1,76 @@
 defmodule Jerboa.Client do
   @moduledoc """
-  STUN client
+  STUN client process
+
+  `start/1` function is used to spawn new client process:
```
Comment (mentels) [inline lib/jerboa/client.ex]: I think that imperative style sounds better: `Use start/1 to ....`

File: lib/jerboa/client.ex [hunk 2]
```diff
@@ -1,28 +1,76 @@
...
+
+      iex> Jerboa.Client.start server: %{address: server_ip, port: server_port}
+      {:ok, #PID<...>}
+
+  (see `start/1` for configuration options)
+
+  Currently the only implemented STUN method is Binding, which allows client to
```
Comment (mentels) [inline lib/jerboa/client.ex]: * Maybe some link explaining the `binding method` ?
* `a client`

File: lib/jerboa/client.ex [hunk 3]
```diff
@@ -1,28 +1,76 @@
...
+  query the server for its external IP address and port. To achieve that, this module
+  provides `bind/1` and `persist/1` functions.
+
+  `bind/1` issues a Binding request to a server and returns reflexive IP address and port.
```
Comment (mentels) [inline lib/jerboa/client.ex]: * The bind/1...

File: lib/jerboa/client.ex [hunk 4]
```diff
@@ -1,28 +1,76 @@
...
+      {{192, 168, 1, 20}, 32780}
+
+  `persist/1` sends a Binding indication to a server, which is not meant to return
+  any response, but is an attempt to refresh NAT bindings in routers on the path to a server.
+  Note that this is only an attempt, there is no guarantee that some router on the path
+  won't rebind client's inside address and port.
   """
 
+  @type t :: pid
+  @type portn :: :inet.port_number
```
Comment (mentels) [inline lib/jerboa/client.ex]: For me the name `portn` is not intuitive. Maybe we could come up with something better? `port_no`? `port_num`?

File: lib/jerboa/client.ex [hunk 5]
```diff
@@ -1,28 +1,76 @@
...
+
+      iex> Jerboa.Client.start server: %{address: {192, 168, 1, 20}, port: 3478}
+      {:ok, #PID<...>}
+
+  ### Options
+
+  * `server` - required - a map with `:address` and `:port` keys
+  """
+  @spec start(options :: Keyword.t) :: Supervisor.on_start_child
   def start(x) do
```
Comment (mentels) [inline lib/jerboa/client.ex:46]: I "love" this kind of naming: `x` :)

File: lib/jerboa/client.ex [hunk 6]
```diff
@@ -1,28 +1,76 @@
...
     Supervisor.start_child(Client.Supervisor, [server(x)])
   end
 
-  @spec bind(GenServer.server) :: {:inet.ip_address, :inet.port_number}
+  @doc """
+  Sends Bindign request to a server
```
Comment (mentels) [inline lib/jerboa/client.ex]: Binding not Bindign

File: lib/jerboa/client.ex [hunk 7]
```diff
@@ -1,28 +1,76 @@
...
+
+  Returns reflexive address and port on successful response. In other case
```
Comment (mentels) [inline lib/jerboa/client.ex]: Otherwise the client process...

File: lib/jerboa/client/worker.ex [hunk 8]
```diff
@@ -13,7 +16,7 @@ defmodule Jerboa.Client.Worker do
 
   def init(address: a, port: p) do
     false = Process.flag(:trap_exit, true)
-    {:ok, socket} = UDP.open(port(), [:binary, active: false])
+    {:ok, socket} = UDP.open(0, [:binary, active: false])
```
Comment (mentels) [inline lib/jerboa/client/worker.ex]: Just plain `0` is really not readable, especially that you alias :gen_udp

**Review by mentels — COMMENTED** [review]
Some changes I required :)

File: lib/jerboa/client.ex (hunk 2 above)
Comment (arkgil) [reply lib/jerboa/client.ex]: Unfortunately I can't find a part explicitly describing the binding method in the RFC.

**Pull request merged**